packages/web-components/fast-foundation/src/disclosure/disclosure.spec.md

```diff
@@ -0,0 +1,98 @@
+# Disclosure
+
+## Overview
+
+A disclosure component is the combination of a button and section of content, a button that toggles the visibility of the section. As defined by the W3C:
+
+> A disclosure is a button that controls the visibility of a section of content. When the controlled content is hidden, it is often styled as a typical push button with a right-pointing arrow or triangle to hint that activating the button will display additional content. When the content is visible, the arrow or triangle typically points down.
+
+### Use Cases
+
+-   In general, reveal the extra information on button click when the end-user is needed to know/see.
+-   Laurel wanted to buy a new phone and she was looking at one of the websites where all the basic information was shown. She read about a camera feature called Night Mode and there was a section labeled "See the advanced technology that goes into every Night mode shot". She clicked on it and found out how that feature would work. With that extra information on that feature, she was quite confident in her buying.
+
+### Features
+
+-   Enter and Space: activates the disclosure component and toggles the visibility of the disclosure content.
+-   Use `expanded` attribute or `toggle()` to render default expanded
+-   Observe the state when the disclosure is toggled
+-   `show()` and `hide()` as helper methods to toggle the content from outside
+
+### Prior Art/Examples
+
+-   [Shopify Polaris](https://polaris.shopify.com/components/behavior/collapsible)
+-   [Lightning Design](https://www.lightningdesignsystem.com/components/expandable-section/)
+-   [Bootstrap](https://getbootstrap.com/docs/4.3/components/collapse/)
+-   [Material UI](https://material-ui.com/api/collapse/)
+
+---
+
+## Design
+
+The disclosure component can easily be extended for customization, for example adding animation to reveal the content or get the height of extra content slot.
+
+-   `showAnimation({ contentNode })` and `hideAnimation({ contentNode })` `async` methods are `protected` and designed to override in the derived class to add animations.
+-   To have further control to the extra content, there are also `contentNode` and `contentHeight` available as getters.
+
+### API
+
+-   _Component Name:_ `fast-disclosure`
+-   _Props/Attrs:_
+    -   `expanded: boolean` - Current state of the disclosure component
+-   _Methods_:
+    -   `show()`: to show the content
+    -   `hide()`: to hide the content
+    -   `toggle()`: to toggle the content
+-   _Events_
+    -   `expanded-changed: CustomEvent` - No custom data.
+
+### Anatomy and Appearance
+
+```html
+<host>
+    <slot name="invoker"></slot>
+    <slot name="content"></slot>
```

Comment:
I think we should have the content go into the default slot so we don't get two named slots and no default slot.

Reply:
Yes, I think that's a good idea.

```html
<host>
    <slot name="invoker"></slot>
    <slot></slot>
</host>
```

Reply:
Done :)